src/Razor/src/Microsoft.AspNetCore.Razor.ExternalAccess.RoslynWorkspace/RazorWorkspaceListenerBase.cs

```diff
@@ -227,18 +228,18 @@ private protected async virtual ValueTask ProcessWorkAsync(ImmutableArray<Work>
 
         cancellationToken.ThrowIfCancellationRequested();
 
-        // Early bail check for if we are disposed or somewhere in the middle of disposal 
+        // Early bail check for if we are disposed or somewhere in the middle of disposal
         if (_disposed || stream is null || solution is null)
```

Comment:
Another thought: Why is checking `stream` and `solution` for null here? They were assigned to locals at the start of the method, and I don't _think_ they can ever be null at this point.

1. `_disposed` is always set to true before `_stream` is set to null. So, `stream` could never be null.
2. `_workspace` is _never_ set to null, and no work is enqueued until `_workspace` is provably non-null. So, `_workspace?.CurrentSolution` can never be null when processing a batch of work.

To clarify the invariants, considering use `AssumeNotNull()` before and after the `_disposed` check to clear up avoid nullability warnings:

```C#
// Capture as locals here. Cancellation of the work queue still need to propogate. The cancellation
// token itself represents the work queue halting, but this will help avoid any assumptions about nullability of locals
// through the use in this function.
var stream = _stream;
var solution = _workspace.AssumeNotNull().CurrentSolution;

cancellationToken.ThrowIfCancellationRequested();

// Early bail check for if we are disposed or somewhere in the middle of disposal
if (_disposed)
{
    _logger.LogTrace("Skipping work due to disposal");
    return;
}

stream.AssumeNotNull();

await CheckConnectionAsync(stream, cancellationToken).ConfigureAwait(false);
```